akka-stream/src/main/scala/akka/stream/scaladsl/StreamConverters.scala

```diff
@@ -111,11 +111,14 @@ object StreamConverters {
    */
   def javaCollector[T, R](collectorFactory: () => java.util.stream.Collector[T, _ <: Any, R]): Sink[T, Future[R]] =
     Flow[T]
-      .fold(() => new CollectorState[T, R](collectorFactory().asInstanceOf[Collector[T, Any, R]])) {
-        (state, elem) => () =>
-          state().update(elem)
+      .fold {
+        new FirstCollectorState[T, R](collectorFactory.asInstanceOf[() => java.util.stream.Collector[T, Any, R]]): CollectorState[
+          T,
```

Comment:
funky formatting

Reply:
The funky formatter stubbornly refuses to let me format it in a sane way.